### PR TITLE
Fix EZP-24389: Implement FieldDefinition form mapper for ezcountry

### PR DESCRIPTION
**Pull request by glye**
https://jira.ez.no/browse/EZP-24389

- [x] Read country data from ezpublish.fieldType.ezcountry.data and show it in the form
- [x] Save default selection
- [x] Test the transformer

### CODE REVIEW

**Comment by glye** [comment]
Review ping @lolautruche @pspanja @andrerom @bdunogier @yannickroger

**Comment by andrerom** [comment]
So the FieldType does not support default selection is that what your saying?
We can create issue for that and handle it separately, as it might affect Platform UI as well ( ? @dpobel )

**Comment by glye** [comment]
Yes, that's what I'm saying - if I'm reading it correctly. Am wondering if the feature may have been deprecated, or just forgotten.

**Comment by lolautruche** [comment]
IMHO it has been forgotten. I guess it would not be very hard to implement this.

**Comment by glye** [comment]
Rebased after ISBN mapper commit.

**Comment by glye** [comment]
@lolautruche @andrerom Legacy reads the country data out of settings/country.ini:
https://github.com/ezsystems/ezpublish-legacy/blob/master/settings/country.ini
We can copy this to somewhere in new stack, or use the Symfony Intl component:
http://symfony.com/doc/current/components/intl.html
Intl holds only country name and 2-letter code. The old country.ini also holds a 3-letter code and phone dialing code, but those don't seem to be used in the ezcountry type. So I think Intl can be used here.

**Comment by lolautruche** [comment]
We already have the data, and it's passed to the FT: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/settings/fieldtypes.yml#L28

**Comment by glye** [comment]
Ah, there it is, thanks.

**Comment by glye** [comment]
Now it reads country data from ezpublish.fieldType.ezcountry.data, but saving doesn't work yet.

File: lib/FieldType/Mapper/CountryFormMapper.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+class CountryFormMapper implements FieldTypeFormMapperInterface
+{
+    /**
+     * @var array
+     */
+    protected $countriesInfo;
+
+    /**
+     * @param array $countriesInfo Array of countries from ezpublish.fieldType.ezcountry.data
```
Comment (lolautruche) [inline lib/FieldType/Mapper/CountryFormMapper.php]: Better move this to property doc instead

File: lib/FieldType/Mapper/CountryFormMapper.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+                [
+                    'required' => false,
+                    'property_path' => 'fieldSettings[isMultiple]',
+                    'label' => 'field_definition.ezcountry.is_multiple',
+                ]
+            )
+            ->add(
+                'defaultSelection',
+                'choice',
+                [
```
Comment (lolautruche) [inline lib/FieldType/Mapper/CountryFormMapper.php]: CS: `[` should be on the same line than the last comma. Makes you save 1 indent

**Comment by glye** [comment]
@lolautruche Fixed feedback on doc and CS, though phpcs complains that "Only one argument is allowed per line in a multi-line function call". Test is TODO.

**Comment by glye** [comment]
Added tests, ready for review

**Comment by lolautruche** [comment]
+1

**Comment by pspanja** [comment]
+1

**Pull request merged**